src/helpers/insights.ts

```diff
@@ -0,0 +1,53 @@
+import { InsightsClientMethod, InsightsClientPayload } from '../types';
+
+export function readDataAttributes(
+  domElement: HTMLElement
+): {
+  method: InsightsClientMethod;
+  payload: InsightsClientPayload;
+} {
+  const method = domElement.getAttribute('data-insights-method');
+  const serializedPayload = domElement.getAttribute('data-insights-payload');
+  if (typeof serializedPayload !== 'string') {
+    throw new Error(`The analytics helper expects payload to be a JSON string`);
```

Comment:
Improve message mentioning `data-insights-payload`. There's not much information on how to fix the problem.

Reply:
changed to :
> The insights helper expects data-insights-payload to be a JSON string